internal/backends/mcu/renderer/draw_functions.rs

```diff
@@ -260,23 +245,54 @@ fn interpolate_color(a: u32, color1: Color, color2: Color) -> Color {
     col
 }
 
-/// Fill (or blend) the color in the buffer
-pub fn blend_buffer(to_fill: &mut [Rgb888], color: Color) {
-    if color.alpha() == u8::MAX {
-        to_fill.fill(super::to_rgb888_color_discard_alpha(color))
-    } else {
-        for pix in to_fill {
-            blend_pixel(pix, color);
+/// Trait for the pixels in the buffer
+pub trait DstPixel: Sized + Copy {
```

Comment:
I find it rather confusing that there's a `DstPixel` trait here and in other places in the same crate `DstPixel` is an alias for a struct.

Maybe the struct could be called `ScreenPixel` (that's the one used finally for the screen) and the trait here could be called just `Pixel` or `PixelOperations`?